pkg/actors/health/health.go

```diff
@@ -0,0 +1,297 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package health
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"sync"
+	"time"
+
+	"k8s.io/utils/clock"
+
+	"github.com/dapr/kit/logger"
+)
+
+const (
+	defaultInitialDelay           = time.Second * 1
+	defaultFailureThreshold       = int32(2)
+	defaultRequestTimeout         = time.Second * 2
+	defaultHealthyStateInterval   = time.Second * 3
+	defaultUnHealthyStateInterval = time.Second / 2
```

Comment:
See my other comment
```suggestion
	defaultUnHealthyStateInterval = time.Millisecond * 1500
```

Reply:
As explained above, 500ms is already a long time for an HTTP endpoint over localhost that does no work, and request timeout covers an overloaded network.